### ISSUE TITLE
ibm_sm_private_certificate unsupported argument: rotation.rotate_keys

### ISSUE THREAD

**Issue by davesteinberg** [IBM-Cloud/terraform-provider-ibm#4976]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s) or Datasource(s)

* ibm_sm_private_certificate

### Description

I tried using `rotation.rotate_keys` as [documented here](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/sm_private_certificate#rotate_keys). Here's my resource:

```
resource "ibm_sm_private_certificate" "vpn_server" {
  instance_id          = ibm_resource_instance.sm.guid
  region               = var.region
  name                 = "${var.prefix}-management-vpn-server-cert"
  certificate_template = resource.ibm_sm_private_certificate_configuration_template.vpn_server.name
  secret_group_id      = ibm_sm_secret_group.common.secret_group_id
  common_name          = "vpn-server.vpn.ibm.com"
  ttl                  = "${365 * 3}d"
  rotation {
    auto_rotate = true
    rotate_keys = true
    interval    = 365 * 3 - 30
    unit        = "day"
  }
}
```

When I try to run `terraform apply` with the latest version of the provider (1.60.0), it fails:

```
$ terraform apply        
╷
│ Error: Unsupported argument
│ 
│   on vpc.tf line 199, in resource "ibm_sm_private_certificate" "vpn_server":
│  199:     rotate_keys = true
│ 
│ An argument named "rotate_keys" is not expected here.
```

If I simply remove `rotate_keys` from the `rotation`, it works.

If this argument isn't supported (yet?), it should be removed from the documentation.

**Comment by ArikShifer** [comment]
Argument is only supported when using an `ibm_sm_public_certificate` resource. It will be removed from `ibm_sm_private_certificate` doc.

**Comment by IdanAdar** [comment]
Fixed in 1.61.
